xdsl/ir/core.py

```diff
@@ -781,6 +761,23 @@ def verify(self, verify_nested_ops: bool = True) -> None:
             if isinstance(operand, ErasedSSAValue):
                 raise Exception("Erased SSA value is used by the operation")
 
+        if (parent_block := self.parent) is not None and (
+            parent_region := parent_block.parent
+        ) is not None:
+            # TODO single-block regions dealt when the NoTerminator trait is
+            # implemented (https://github.com/xdslproject/xdsl/issues/1093)
+            if len(parent_region.blocks) > 1:
```

Comment:
In the case where there is a single block, we still need to check that an operation that is not the last does not have a successor (with or without `NoTerminator`), right?
Or is it different since you technically are not allowed to jump to the entry block?

Reply:
That is right but I was thinking it might be simpler to check for both cases when the `NoTerminator` trait is implemented. I guess the branching off for a non-last op is irrelevant to that, so we can just test it now.

Am I understanding correctly?

Reply:
I changed my mind on this, it's simpler and more clear to check that an operation should not have successors unless it is the last operation in its parent block (regardless of the number of blocks in the parent region or the existence of the `NoTerminator` trait.